src/App.js

```diff
@@ -212,22 +214,46 @@ function App() {
     return urlParams.has('mode') && urlParams.get('mode') === 'showLocationSummary';
   }
 
-  function getLocationParam() {
+  function getParam(param) {
     const urlParams = new URLSearchParams(window.location.search);
-    return urlParams.get('location');
+    return urlParams.get(param);
   }
 
   function showLocationSummary() {
-    const location = getLocationParam();
-    const attemptsForLocation = attempts.filter((attempt) => {
-      return attempt.selectedExample === location;
-    });
-    if (attemptsForLocation.length > 0) {
-      const lastAttempt = attemptsForLocation[attemptsForLocation.length - 1];
-      return <LocationSummary attempt={lastAttempt} location={location} />;
-    } else {
+    if (!attemptsLoaded) {
       return null;
     }
+    const location = getParam('location');
+    const attemptsForLocation = getAttemptsForLocation(location);
+    let lastAttempt = null;
+    if (attemptsForLocation.length > 0) {
+      lastAttempt = attemptsForLocation[attemptsForLocation.length - 1];
+    }
+    return showLocationAttempt(lastAttempt, location);
+  }
+
+  function getAttemptsForLocation(location) {
+    let locationAttempts = [];
+    if (attempts) {
+      locationAttempts = attempts.filter((attempt) => {
+        return attempt.selectedExample === location;
+      });
+    }
+    return locationAttempts;
+  }
+
+  function showLocationAttempt(lastAttempt, location) {
+    const showLocationText = getParam('showLocationText') === 'false' ? false : true;
```

Comment:
This tertiary check is a bit hard to read. How about this?
```
const showLocationText = getParam('showLocationText') !== 'false';
```

Reply:
Good suggestion, thanks. I updated.